### ISSUE TITLE
Relative/absolute paths and running Hector

### ISSUE THREAD

**Issue by bpbond** [JGCRI/hector#541]
When I run from Xcode (or the command line) I definitely get output files, but @kdorheim and others do not. Why? Fix. This is come up before in previous issues—lazy right now, not searching, but they're there.

**Comment by pralitp** [comment]
My first thought without looking into this too much is that there are no checks for if the output directory does not already exist.  I don't _think_ it will create it for you -- or maybe it was a platform specific deal.

But you can check logger to see how we dealt with it for the logs directory:
https://github.com/JGCRI/hector/blob/a75702ba5a3ab9684a4eae1e49ef221172b6a494/src/logger.cpp#L272

Maybe if this is indeed the issue we an move that util function some place else and use it to check the `OUTPUT_DIRECTORY` before trying to open the output files for writing.

**Comment by kdorheim** [comment]
Last time checked even when I manually make the output directory and run Hector via command line or Xcode no csv file is written out, but let me try again.

**Comment by bpbond** [comment]
Is the problem that if the output directory doesn't exist, it won't get created? What I understand from you @kdorheim is that your Hector isn't writing _any_ outputs, even when the output directory is there.

**Comment by kdorheim** [comment]
Thanks @bpbond and @pralitp for taking a look at this. 

From the command line the output directory has to exist in order for the output csv files to be written there. By including read me in the output directory in #543 that directory will always exist. 

For Xcode the schemes need to be set up with the correct working directory in order to be able to write the csv files out, #543 adds more documentation about setting up the correct path and the working directory for the scheme. 

So when #543 is merged this issue can be closed?

**Comment by bpbond** [comment]
@kdorheim I think so!

**Comment by pralitp** [comment]
I still think we should at the very least have some error checking that in main.cpp:68
```
        // Open the stream output file, which has an optional run name (specified in the INI file) in it
        string rn = core.getRun_name();
        if( rn == "" )
            csvoutputStreamFile.open( string( string( OUTPUT_DIRECTORY ) + "outputstream.csv" ).c_str(), ios::out );
        else
            csvoutputStreamFile.open( string( string( OUTPUT_DIRECTORY ) + "outputstream_" + rn + ".csv" ).c_str(), ios::out );
```

Actually opens correctly.  There is some chance a user will try to set some random working directory and forget to create the output dir.  It would be good to catch that.  I think it should be as simple as checking:
```
if(!csvoutputStreamFile.is_open()) {
    // H_THROW here, maybe some refactoring so we can print the full output stream file name we tried to open but failed
}
```

I can take this on in the hackathon if you think it is a good idea.

**Comment by kdorheim** [comment]
^^ @pralitp that is a great idea and that would be really helpful thanks!

**Comment by pralitp** [comment]
Oops, I think this is a duplicate of #115 which is already part of the `v3.0` milestone so I'm just gonna close this one.
